File: src/main/java/com/chrisdempewolf/responses/pin/PinResponse.java
package com.chrisdempewolf.responses.pin;

public class PinResponse {
    private final Pin pin;

    public PinResponse(Pin pin) {
        this.pin = pin;
    }

    public Pin getPin() {
        return pin;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        PinResponse pinResponse = (PinResponse) o;

        return !(pin != null ? !pin.equals(pinResponse.pin) : pinResponse.pin != null);
    }

    @Override
    public int hashCode() {
        return pin != null ? pin.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "PinResponse{" +
                "pin=" + pin +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/PinResponse.kt
package com.chrisdempewolf.pinterest.responses.pin

import com.google.gson.annotations.SerializedName

class PinResponse(@SerializedName("data")
                  val pin: Pin?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val pinResponse = o as PinResponse?

        return !if (pin != null) pin != pinResponse!!.pin else pinResponse!!.pin != null
    }

    override fun hashCode(): Int {
        return if (pin != null) pin.hashCode() else 0
    }

    override fun toString(): String {
        return "PinResponse{" +
                "pin=" + pin +
                '}'
    }
}
